specs/data_structures.md

```diff
@@ -105,23 +105,29 @@ Blocks are the top-level data structure of the LazyLedger blockchain.
-| `proposerAddress`                 | [Address](#address)       | Address of this block's proposer.                                                                                                                                                                    |
+| name                              | type                      | description                                                                                                                                                                        |
+| --------------------------------- | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `height`                          | [Height](#type-aliases)   | Block height. The genesis block is at height `1`.                                                                                                                                  |
+| `timestamp`                       | [Timestamp](#timestamp)   | Timestamp of this block.                                                                                                                                                           |
+| `lastBlockID`                     | [BlockID](#blockid)       | Previous block's ID.                                                                                                                                                               |
+| `lastCommitRoot`                  | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                                                                                                                           |
+| `consensusRoot`                   | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block.                                                                                                       |
+| `stateCommitment`                 | [HashDigest](#hashdigest) | The [state root](#state) after this block's transactions are applied.                                                                                                              |
+| `availableDataOriginalSharesUsed` | `uint64`                  | The number of shares used in the [original data square](#arranging-available-data-into-shares) that are not [tail padding](./consensus.md#reserved-namespace-ids).                 |
+| `availableDataOriginalSquareSize` | `uint64`                  | The number of rows/columns of the original data [shares](data_structures.md#share) in [square layout](#arranging-available-data-into-shares) for this block. Must be a power of 2. |
```

Comment:
Why do we still need this field, isn't there a one-to-one mapping between availableDataOriginalSharesUsed and availableDataOriginalSquareSize?

Reply:
`availableDataOriginalSharesUsed` is the number of shares used that aren't tail padding (up to `availableDataOriginalSquareSize ** 2`), and `availableDataOriginalSquareSize` is the size of the original square.

Reply:
My point is that for any given value of `availableDataOriginalSharesUsed`, it seems to me that given value should always have the same `availableDataOriginalSquareSize`, so `availableDataOriginalSquareSize` can just be recomputed from `availableDataOriginalSharesUsed`.

Reply:
Assuming we have a rule that the square size must always be as small as possible, I think that's right. But we don't yet have such a rule. Do we want one?

Reply:
If we remove availableDataOriginalSquareSize, that will be implicit, yet.

Reply:
Removed `availableDataOriginalSquareSize` in 3a5c731. cc @liamsi